examples/gitserver/README.md

```diff
@@ -1,16 +1,207 @@
+
+   ```sh
+   $ git config --global http.https://gitserver-myproject.infra.openshift.com.sslCAInfo /path/to/ca.crt
+   ```
+
+   where the key is http.[git server URL].sslCAInfo
+
+3. Disable anonymous cloning. By default the gitserver will allow anonymous cloning to make it easier to 
+   run builds without having to specify a secret. You can disable this by setting the `ALLOW_ANON_GIT_PULL`
+   environment variable to `no`.
+
+   ```sh
+   $ oc set env dc/git ALLOW_ANON_GIT_PULL=no
+   ```
+
+   **NOTE:** changing environment variables on the git server will cause a redeploy of the git server. If using
+   the default ephemeral storage for it, all repositories that have been pushed previously will be wiped out.
+   They will need to be pushed to the server again to restore.
+
+Authentication
+--------------
+
+By default, the gitserver will authenticate using OpenShift user or service account credentials. For a user,
+the credentials are the user name, and the user's token (from `oc whoami -t`). For a service account, it can
+be one of the 2 tokens created with the service account and stored in the service account secrets. These can 
+be obtained by looking at the secrets that correspond to the service account (`oc get secrets`) and displaying
+the token in one of them: `oc describe secret secret-name`.
+
+
+Authorization
+-------------
+
+Users or service accounts must be able to get pods in the current namespace in order to fetch repositories from
+the Git Server. Specifying ALLOW_ANON_GIT_PULL=yes as an environment variable for the Git Server will allow anyone
+to fetch/clone content from the Git Server. To create/push content, users or service accounts must have the right 
+to create pods in the namespace.
+
+
+Automatically Cloning Public Repositories
+-----------------------------------------
+
+The Git Server can automatically clone a set of repositories on startup if it is going to be used for mirroring
+purposes. Repositories to be cloned can be specified using a set of environment variables that match
+`GIT_INITIAL_CLONE_[name]` where [name] must be unique. The value of the variable must be a URL to the remote
+repository to clone and an optional name for the clone.
+
+The following command will add an environment variable to clone the openshift/ruby-hello-world repository and give
+it a name of `helloworld` inside the Git Server.
+
+```sh
+$ oc set env dc/gitserver GIT_INITIAL_CLONE_1="https://github.com/openshift/ruby-hello-world.git;helloworld"
+```
+
+Automatically Starting Builds
+-----------------------------
+
+By default, whenever the git server receives a commit, it will look for a BuildConfig named the same as the
+repository where the commit is being pushed. If it finds a BuildConfig with the same name, it will start a build
+for that BuildConfig. Alternatively, an annotation may be added to a BuildConfig to explicitly link it to a 
+repository on the git server:
```

Comment:
will it start builds for all matching buildconfigs, or just the first one?  (suppose i add the annotation to multiple BCs)